### ISSUE TITLE
TypeError: Cannot read properties of undefined (reading 'total')

### ISSUE THREAD

**Issue by AlexMuia31** [adrianhajdin/project_cryptoverse#17]
×
TypeError: Cannot read properties of undefined (reading 'total')
Homepage
src/components/Homepage.jsx:23

> 23 |     <Col span={12}><Statistic title="Total Cryptocurrencies" value={globalStats.total} /></Col>
That's an erroe in HomePage.jsx

What could be the solution? Even the hosted project doesn't load

**Comment by AlexMuia31** [comment]
The coinranking API is down...status code 503

**Comment by arasouli91** [comment]
They are migrating to V2 api. It was planned to be down today. I am trying to use the temporary V2 migration api and it works fine just putting this in the browser: https://api.coinranking.com/v2/migration/coins
But, in my app I am getting this response:
![image](https://user-images.githubusercontent.com/19616536/141039550-3e1549fd-902e-4d3f-a06f-c27a25a68898.png)

**Comment by Arif-arsh** [comment]
Same issue have solved it?

**Comment by AlexMuia31** [comment]
Yes right now the API is back up and running

**Comment by Arif-arsh** [comment]
Still I am getting the same error how to fix it?

**Comment by Arif-arsh** [comment]
I am getting undefined (reading 'total')

**Comment by AlexMuia31** [comment]
Do you have this part?
if (isFetching) return "Loading . . .";

**Comment by Arif-arsh** [comment]
yes I have it still getting the same error

**Comment by aindriu80** [comment]
I'm having the undefined error as well... there seems to be some kind of API error with RapidAPI and the returned object. 

There are several posts on StackOverflow about using the Fetch method but I couldn't get it to work.

**Comment by shubhamparmar8** [comment]
You need to migrate to their new API v2. Few changes like coin uuid in place of id, supply property in cryptoDetails object do read their docs about migrating to v2. Exchange data endpoint now needs pro plan or higher. 
https://developers.coinranking.com/api/documentation

**Comment by ParkerM2** [comment]
Has anyone had any luck getting this to work?

**Comment by Arif-arsh** [comment]
Add a question mark before the total.
Something?.total
it worked for me!

**Comment by ishitamittal1210** [comment]
> Add a question mark before the total. Something?.total it worked for me!

Yeah by doing this I am able to see my window but value displayed in front of Total Cryptocurrencies is 0.How to resolve this?

**Comment by ParkerM2** [comment]
> > Add a question mark before the total. Something?.total it worked for me!
> 
> Yeah by doing this I am able to see my window but value displayed in front of Total Cryptocurrencies is 0.How to resolve this?

I have it working on mine if you want to look at the code here -> https://github.com/ParkerM2/Crypto-Site/blob/main/src/components/HomePage.jsx

**Comment by someshkrish** [comment]
I think you guys are discussing bout this  ```globalStats.total```.

My opinion:
- check whether the api returned the response properly by doing ```console.log(globalStats)```
